src/utils/intl.js

```diff
@@ -0,0 +1,52 @@
+import reactCookie from 'react-cookie';
+import { locale as defaultLocale, locales } from 'config';
+
+const maxAge = 24 * 60 * 60 * 1000 * 7;
+const getDefaultLocale = () => {
+  const localeFromCookie = reactCookie.load('locale');
+  return locales.indexOf(localeFromCookie) !== -1 ? localeFromCookie : defaultLocale;
+};
+const IntlUtils = {
+
+  intlDataHash: {
+    en: {
+      file: 'en',
+      locale: 'en'
+    },
+    fr: {
+      file: 'fr',
+      locale: 'fr'
+    },
+  },
+  initializer(params, history, location, loader) {
+
+    let currentLocale = getDefaultLocale();
+
+    if (params.locale && locales.indexOf(params.locale) !== -1) {
+      currentLocale = params.locale;
+    } else {
+      setTimeout(() => {
+        history.replaceState(null, `/${currentLocale}${location.pathname}`);
+      });
+    }
```

Comment:
Maybe I'm missing where is done in this PR. But I do not see if you are detecting user's browser locale from server headers. [I've done this in rails](http://guides.rubyonrails.org/i18n.html#setting-the-locale-from-the-client-supplied-information): 

``` ruby
user_locale = extract_locale_from_accept_language_header
```

And again. Maybe this logic could be done on `onEnter` hook of react-router